buildpacks/metrics-agent/src/layers/install_agentmon.rs

```diff
@@ -0,0 +1,329 @@
+                else
+                    echo "To enable logging run with AGENTMON_DEBUG=1" >> {log_file}
+
+                    start-stop-daemon --start --background \
+                        --exec "{agentmon_loop}" \
+                        -- --path {agentmon_path}
+                fi
+            "#
+            ),
+        )
+        .map_err(DownloadAgentmonError::CouldNotWriteDestinationFile)?;
+
+        execd_script
+    };
+
+    Ok(execd_script)
+}
+
+#[cached]
+fn get_latest_url() -> Result<Url, GetUrlError> {
+    // This file on S3 stores a raw string that holds the URL to the latest agentmon release
+    // It's not a redirect to the latest file, it's a string body that contains a URL.
+    let base = Url::parse("https://agentmon-releases.s3.amazonaws.com/latest")
+        .expect("Internal error: Bad url");
+
+    let body = ureq::get(base.as_ref())
+        .call()
+        .map_err(|err| GetUrlError::RequestError(err.to_string()))?
+        .into_string()
+        .map_err(|error| GetUrlError::CannotConvertResponseToString(error.to_string()))?;
+
+    Url::parse(body.as_str().trim()).map_err(GetUrlError::UrlParseError)
+}
+
+fn download_to_dir(destination: &Path, url: &Url) -> Result<(), DownloadAgentmonError> {
+    let agentmon_tgz =
+        NamedTempFile::new().map_err(DownloadAgentmonError::CouldNotCreateDestinationFile)?;
+
+    download(url.as_ref(), agentmon_tgz.path())?;
+
+    untar(agentmon_tgz.path(), destination)?;
+
+    chmod_plus_x(&destination.join("agentmon")).map_err(DownloadAgentmonError::PermissionError)?;
+
+    Ok(())
+}
+
+pub(crate) fn untar(
+    path: impl AsRef<Path>,
+    destination: impl AsRef<Path>,
+) -> Result<(), DownloadAgentmonError> {
+    let file =
+        fs_err::File::open(path.as_ref()).map_err(DownloadAgentmonError::CouldNotOpenFile)?;
+
+    Archive::new(GzDecoder::new(file))
+        .unpack(destination.as_ref())
+        .map_err(DownloadAgentmonError::CouldNotUnpack)
+}
+
+/// Sets file permissions on the given path to 7xx (similar to `chmod +x <path>`)
+pub fn chmod_plus_x(path: &Path) -> Result<(), std::io::Error> {
+    let mut perms = fs_err::metadata(path)?.permissions();
+    let mut mode = perms.mode();
+    octal_executable_permission(&mut mode);
+    perms.set_mode(mode);
+
+    fs_err::set_permissions(path, perms)
+}
+
+/// Write a script to the target path while adding a bash shebang line and setting execution permissions
+fn write_bash_script(path: &Path, script: impl AsRef<str>) -> std::io::Result<()> {
+    let script = script.as_ref();
+    fs_err::write(path, format!("#!/usr/bin/env bash\n\n{script}"))?;
+    chmod_plus_x(path)?;
+
+    Ok(())
+}
+
+/// Ensures the provided octal number's executable
+/// bit is enabled.
+///
+/// i.e. chmod +x will ensure that the first digit
+/// of the file permission is 7 on unix so if you pass
+/// in 0o455 it would be mutated to 0o755
+fn octal_executable_permission(mode: &mut u32) {
+    *mode |= 0o700;
+}
```

Comment:
This function just wraps a bitwise OR and has exactly one call-site. If we just inline the `|=` we can remove all that extra machinery, docs and tests. 

The docs for this function and `chmod_plus_x` make that even more evident since they both reference each others responsibility and blur into one. 

If we think we need to comment how bitwise operations work (I personally don't think we do) we can add a comment above the `|=` too.

Reply:
I've decided to inline it as ssuggested and add an extra assertion to the existing test:

```rust
        assert_eq!(after, after | 0o700);
```